### PR TITLE
#3842 expose Ohara's partitioner through File APIs

### PR DESCRIPTION
**Pull request by chia7712**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #3842)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-common/src/main/java/com/island/ohara/common/setting/WithDefinitions.java [hunk 1]
```diff
@@ -83,16 +85,45 @@ static SettingDef revisionDefinition(String revision) {
...
+        .permission(SettingDef.Permission.READ_ONLY)
+        .build();
+  }
+
+  enum Type {
+    SOURCE,
+    SINK,
+    PARTITIONER,
+    STREAM,
+    UNKNOWN;
```
Comment (jackyoh) [inline ohara-common/src/main/java/com/island/ohara/common/setting/WithDefinitions.java]: 這裡怎麼沒有 shabondi

Comment (chia7712) [reply ohara-common/src/main/java/com/island/ohara/common/setting/WithDefinitions.java]: 對ㄟ 我補上去

Comment (jackyoh) [reply ohara-common/src/main/java/com/island/ohara/common/setting/WithDefinitions.java]: Thank you.

**Comment by chia7712** [comment]
retry

**Comment by chia7712** [comment]
retry testing-util

**Comment by chia7712** [comment]
retry streams

**Comment by chia7712** [comment]
retry configurator

**Comment by chia7712** [comment]
retry agent

**Comment by chia7712** [comment]
retry client

**Comment by chia7712** [comment]
retry connector

**Comment by chia7712** [comment]
retry manager-api

**Comment by chia7712** [comment]
retry manager-ut

**Comment by chia7712** [comment]
retry shabondi

**Comment by chia7712** [comment]
retry common

**Comment by chia7712** [comment]
retry metrics

**Comment by chia7712** [comment]
retry connector

**Comment by chia7712** [comment]
retry kafka

**Comment by chia7712** [comment]
our jenkins is flaky and I have tested the remaining tests on my local.

**Pull request merged**